packages/in_app_purchase/android/src/main/java/io/flutter/plugins/inapppurchase/InAppPurchasePlugin.java

```diff
@@ -181,7 +195,38 @@ private void launchBillingFlow(String sku, @Nullable String accountId, Result re
     result.success(billingClient.launchBillingFlow(activity, paramsBuilder.build()));
   }
 
-  private void updateCachedSkus(List<SkuDetails> skuDetailsList) {
+  private void queryPurchases(String skuType, Result result) {
+    if (billingClientError(result)) {
+      return;
+    }
+
+    // Like in our connect call, consider the billing client responding a "success" here regardless of status code.
+    result.success(fromPurchasesResult(billingClient.queryPurchases(skuType)));
```

Comment:
Just want to make sure the response object that returned by `billingClient.queryPurchases(skuType)` should never be null, correct?

Reply:
Yes, this should always be non-null.